### DISCUSSION TITLE
Supabase Twitter oauth is logging in as the same user? Same UID every time.

### DISCUSSION THREAD

**Discussion by richiemcilroy** [supabase/supabase#2057] [Questions]
I've implemented Supabase into Next.js, got auth to work, but when logging in via Twitter it's logging into the same UID every single time, when the twitter account is different. It provides the exact same UID every time.

Initial function:

```
  const handleOAuthSignIn = async (provider) => {
    setLoading(true);
    const { error } = await signIn({ provider });
    if (error) {
      setMessage({ type: 'error', content: error.message });
    }
    setLoading(false);
  };
```

Sign in request:

`signIn: (options) => supabase.auth.signIn(options),`

Any ideas?

**Comment by silentworks** [comment]
Can you check what comes back in data when you sign in please?

```js
const { error, data } = await signIn({ provider });
console.log({ data });
```

Also once you are logged in, in another part of the app check what session returns

```js
const session = supabase.auth.session();
console.log({ session });
```

If these are returning the correct UUID, then it means it's something else in your app that's causing this.
